spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phrase, such as dev, testing, live, etc. 
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice, e.g.
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile%`, (e.g. `%live%vechile.name`).
+
+For an instance, a config source can be specified as follows.
+
+[source, text]
+----
+%dev%vehicle.name=car
+%live%vehicle.name=train
+%testing%vehicle.name=bike
+----
+
+In the above config source, if the system property `mp.config.profile` is set to `dev`. The `vechileName` in the following snippet will be resolved to be `car`.
+Similarly, it will be `train` if the `mp.config.profile` is set to `live`, `bike` if the profile is `testing`.
+
+[source, text]
+----
+@Inject @ConfigProperty(name="vehicle.name") String vechileName;
+----
+
```

Comment:
We need some normative text covering what conforming implementations are required to do.

Reply:
Detailed description of resolution rules and interplay between ordinals and other config sources are needed here and these rules also tested within the TCK.

Reply:
+1 @smillidge , I will add more TCKs for the config source ordinal rules once we are happy with the rules. See here for more explaination.